File: app/src/main/java/com/veganbeauty/app/data/local/OrderConverters.java
package com.veganbeauty.app.data.local;

import androidx.room.TypeConverter;
import com.veganbeauty.app.data.local.entities.OrderEntity.OrderItem;
import org.json.JSONArray;
import org.json.JSONObject;

import java.util.ArrayList;
import java.util.List;

public class OrderConverters {
    @TypeConverter
    public String fromItemList(List<OrderItem> items) {
        if (items == null) return "[]";
        JSONArray array = new JSONArray();
        try {
            for (OrderItem item : items) {
                JSONObject obj = new JSONObject();
                obj.put("productId", item.getProductId());
                obj.put("productName", item.getProductName());
                obj.put("productImage", item.getProductImage());
                obj.put("quantity", item.getQuantity());
                obj.put("price", item.getPrice());
                array.put(obj);
            }
        } catch (Exception e) {
            e.printStackTrace();
        }
        return array.toString();
    }

    @TypeConverter
    public List<OrderItem> toItemList(String value) {
        if (value == null || value.isEmpty()) return new ArrayList<>();
        List<OrderItem> list = new ArrayList<>();
        try {
            JSONArray array = new JSONArray(value);
            for (int i = 0; i < array.length(); i++) {
                JSONObject obj = array.getJSONObject(i);
                OrderItem item = new OrderItem();
                item.setProductId(obj.getString("productId"));
                item.setProductName(obj.getString("productName"));
                item.setProductImage(obj.getString("productImage"));
                item.setQuantity(obj.getInt("quantity"));
                item.setPrice(obj.getLong("price"));
                list.add(item);
            }
        } catch (Exception e) {
            e.printStackTrace();
        }
        return list;
    }
}

File: app/src/main/java/com/veganbeauty/app/data/local/OrderConverters.kt
package com.veganbeauty.app.data.local

import androidx.room.TypeConverter
import com.veganbeauty.app.data.local.entities.OrderItem
import org.json.JSONArray
import org.json.JSONObject

class OrderConverters {
    @TypeConverter
    fun fromItemList(items: List<OrderItem>?): String {
        if (items == null) return "[]"
        val array = JSONArray()
        for (item in items) {
            val obj = JSONObject()
            obj.put("productId", item.productId)
            obj.put("productName", item.productName)
            obj.put("productImage", item.productImage)
            obj.put("quantity", item.quantity)
            obj.put("price", item.price)
            array.put(obj)
        }
        return array.toString()
    }

    @TypeConverter
    fun toItemList(value: String?): List<OrderItem> {
        if (value.isNullOrEmpty()) return emptyList()
        val list = mutableListOf<OrderItem>()
        val array = JSONArray(value)
        for (i in 0 until array.length()) {
            val obj = array.getJSONObject(i)
            list.add(
                OrderItem(
                    productId = obj.getString("productId"),
                    productName = obj.getString("productName"),
                    productImage = obj.getString("productImage"),
                    quantity = obj.getInt("quantity"),
                    price = obj.getLong("price")
                )
            )
        }
        return list
    }
}
